toolchain/check/testdata/namespace/fail_decl_in_alias.carbon

```diff
@@ -0,0 +1,38 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+
+namespace NS;
+
+alias ns = NS;
+
+// CHECK:STDERR: fail_decl_in_alias.carbon:[[@LINE+6]]:7: ERROR: Name qualifiers are only allowed for entities that provide a scope.
+// CHECK:STDERR: fn ns.A() -> i32 { return 0; }
+// CHECK:STDERR:       ^
+// CHECK:STDERR: fail_decl_in_alias.carbon:[[@LINE+3]]:4: Non-scope entity referenced here.
+// CHECK:STDERR: fn ns.A() -> i32 { return 0; }
+// CHECK:STDERR:    ^~
+fn ns.A() -> i32 { return 0; }
```

Comment:
Why is this an error?

Reply:
Stemming from some recollections in explorer discussion, aliases are not supported when declaring names. This is WAI even if the error isn't easy to understand.

Reply:
Could you add a comment explaining then, please, since this wasn't what I expected? (No need to change the semantics until we have a spec for what `alias` does.)

Reply:
Done